### DISCUSSION TITLE
How to get a imput from the keyboard whyle a program is running

### DISCUSSION THREAD

**Discussion by Vexxen1** [cc-tweaked/CC-Tweaked#1271] [Q&A]
I have a whyle loop. I want to do something when the person presses "k", then once then press k it will pause the whyle loop, and open a menu where they can imput in things. But when k is not pressed i still want it to run the entire whyle loop. 

So i need: 
1. How to get a bollene value of when a key is pressed without waiting for it to be pressed
2. How to wait for imput from the terminal. Like wait for a y/n.

**Comment by SquidDev** [comment]
So the easiest thing to do here is use the parallel API to pull the `key` event while your main loop is running. So something like 

```lua
local running = true
parallel.waitForAll(function()
 while running do 
   -- ... 
  end 
end, function()
  while true do 
    local event, key = os.pullEvent("key")
    if key == keys.k then running = false end 
  end 
end)
```
